lang/syntax.md

```diff
@@ -6,7 +6,7 @@ First, the general structure of programs and functions:
 ```rust
 /// Some opaque type of function names.
 /// The details of this this is represented to not matter.
-type FnName;
+struct FnName(specr::Name);
```

Comment:
I preferred this representation for now, but we can always change it to `#[specr::name] type FnName` or something shall we need it.